bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/troubleshoot/StrandDump.java

```diff
@@ -18,59 +18,114 @@
 
 package io.ballerina.runtime.internal.troubleshoot;
 
-import io.ballerina.runtime.internal.scheduling.Strand;
+import com.sun.management.HotSpotDiagnosticMXBean;
 
+import javax.management.MBeanServer;
+import java.io.File;
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.nio.file.Files;
+import java.nio.file.Paths;
 import java.time.LocalDateTime;
 import java.time.format.DateTimeFormatter;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
 
 /**
  * Used to get the status of current Ballerina strands.
  *
  * @since 2201.2.0
  */
 public class StrandDump {
+    private static final String hotSpotBeanName = "com.sun.management:type=HotSpotDiagnostic";
+    private static volatile HotSpotDiagnosticMXBean hotSpotDiagnosticMXBean;
 
     public static String getStrandDump() {
-        Map<Integer, Strand> availableStrands = new HashMap<>();
-        int createdStrandGroupCount = 0;
-        int createdStrandCount = 0;
-        int availableStrandCount = 0;
-        Map<Integer, List<String>> availableStrandGroups = new HashMap<>();
-
-        String strandDumpOutput = generateOutput(availableStrandGroups, availableStrandCount, createdStrandGroupCount,
-                createdStrandCount);
-        cleanUp(availableStrands, availableStrandGroups);
-        return strandDumpOutput;
+        String filename = "thread_dump" + LocalDateTime.now();
+        String dump;
+        try {
+            getStrandDump(System.getProperty("user.dir") + "/" + filename);
```

Comment:
We can move to constant System.getProperty("user.dir")

Reply:
Fixed with 80fe428001ed625ade093870c89298c84dafa9c3